### PR TITLE
feat(CTA-28) Update User

### PR DESCRIPTION
**Pull request by drallo22**
## What is the ticket about and why are we doing this change.
Implemented the functionality for an Admin to edit a client's information
## What are the various changes and what other modules do those changes effect.
- Created service method and implemented it
- Created Page for update and styled it
- Added edit button to accounts page for each user
- Tested front end
- Tested Backend
## Before and After UI (Required for UI-impacting PRs)

# Before
image

# After
Screenshot 2023-12-16 135918

Screenshot 2023-12-16 135939
Screenshot 2023-12-16 135949

### CODE REVIEW

File: coupetonarbrefrontend/cypress/e2e/DeleteClientByClientId.cy.js [hunk 1]
```diff
@@ -49,14 +49,12 @@ describe('FirstTest', () => {
 
       // Wait for the intercepted request and stub the response
       cy.wait('@deleteRequest').then((interception) => {
-        expect(interception.response.statusCode).to.equal(200); // Assuming a successful deletion
+        expect(interception.response.statusCode).to.equal(204); // Assuming a successful deletion
```
Comment (Matthewtan9) [inline coupetonarbrefrontend/cypress/e2e/DeleteClientByClientId.cy.js:52]: Why did you change 200 to 204, why was it necessary and what is the difference between them.

**Review by Matthewtan9 — APPROVED** [review]
Good job on making it work!

File: coupetonarbrefrontend/cypress/e2e/DeleteClientByClientId.cy.js (hunk 1 above)
Comment (drallo22) [reply coupetonarbrefrontend/cypress/e2e/DeleteClientByClientId.cy.js:52]: 204 is the status code that is used when the server successfully processes the request, but doesn't return anything to the client, so that is the code that is used for delete. I changed it as the test was failing because it was getting a 204 as it should but the test was expecting a 200.

File: coupetonarbrebackend/src/test/java/com/example/coupetonarbrebackend/User/PresentationLayer/UserControllerIntegrationTest.java [hunk 2]
```diff
@@ -68,6 +67,43 @@ void getClientById_shouldSucceed() {
...
+                .expectBody(ClientResponseDTO.class)
+                .value(clientResponseDTO -> {
+                    assert clientResponseDTO != null;
+                    assert clientResponseDTO.getClientId().equals(clientId);
+                    assert clientResponseDTO.getFirstName().equals("John");
+                });
+    }
+
+        // Act & Assert
+
```
Comment (ebond07) [inline coupetonarbrebackend/src/test/java/com/example/coupetonarbrebackend/User/PresentationLayer/UserControllerIntegrationTest.java:106]: This comment is misleading, the act and assert is right before it

File: coupetonarbrefrontend/cypress/e2e/UpdateClient.cy.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    cy.get('input[name="firstName"]').clear().type('UpdatedFirstName');
+    // Add more commands to update other fields if needed
+
+    // Submit the form
+    cy.get('button:contains("Update Client")').click();
+
+    // Verify that the client information has been updated
+    cy.url().should('include', '/accounts'); // Assuming successful update redirects to the Accounts page
+    cy.get('td:contains("UpdatedFirstName")').should('exist'); // Adjust based on your updated data
+    // Add more assertions for other fields if needed
```
Comment (ebond07) [inline coupetonarbrefrontend/cypress/e2e/UpdateClient.cy.js:25]: Make sure to delete these comments that you added as I don't believe you will be adding other fields.

File: coupetonarbrefrontend/src/App.js [hunk 4]
```diff
@@ -1,11 +1,12 @@
...
-import QuotesAdmin from './components/QuotesAdmin';
-import FeedbackAdmin from './components/FeedbackAdmin';
-import Profile from './components/Profile';
+import HomePage from './pages/HomePage';
+import AccountsAdmin from './pages/AccountsAdmin';
+import AppointmentsAdmin from './pages/AppointmentsAdmin';
+import QuotesAdmin from './pages/QuotesAdmin';
+import FeedbackAdmin from './pages/FeedbackAdmin';
+import Profile from './pages/Profile';
+import UpdateClientAdmin from './pages/UpdateClientAdmin';
```
Comment (ebond07) [inline coupetonarbrefrontend/src/App.js:9]: Good job taking the initiative to update the structure to follow better practices.

**Review by ebond07 — APPROVED** [review]
Good job, just don't forget to remove unnecessary comments.

**Review by ebond07 — APPROVED** [review]
Thanks for hearing me out. Also good catch on that empty ClientController.

**Review by Matthewtan9 — APPROVED** [review]
